### DISCUSSION TITLE
Lambda Order of Operations

### DISCUSSION THREAD

**Discussion by mountainbiker54** [CircuitSetup/Expandable-6-Channel-ESP32-Energy-Meter#96] [General]
What's the order of operations on the lambda command.

Is it adding all the values then returning a positive? Or does it make every value positive before adding?

ct1=10
ct2=10
ct3= -5

Is the returned value 15 or 25?

```
#Total Watts Main
  - platform: template
    name: ${disp_name} Total Watts Main
    id: totalWattsMain
    lambda: return id(ct1Watts).state + id(ct2Watts).state + id(ct3Watts).state;
    accuracy_decimals: 1
    unit_of_measurement: W
    device_class: power
    update_interval: ${update_time}
```

**Comment by CircuitSetup** [comment]
The returned value should be 15 since there isn't anything making the values positive within the lambda. To do that you could do something like: 
`lambda: return abs(id(ct1Watts).state) + abs(id(ct2Watts).state) + abs(id(ct3Watts).state);`

Also, if you're getting negative watts when you shouldn't be, the CT is likely backwards on the wire.

**Comment by mountainbiker54** [comment]
Thanks. I'm trying to get rid of the transient negative values when nothing is running on the circuit. [img](https://mike.generalllc.net/projects/powermon/Image%201447.png) I saw thread #87 where you gave the same advice. 

While going through and checking every single one of my 36 clamps sounds like a fun endeavor, it seem like throwing an abs() in front of things seems more fool proof. I was also thinking of an if statement to ignore all the values lower than 1 watt but I have not gotten that far.
